File: LeetCode/Medium/2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        cnt_num = [0] * 201
        for num in nums:
            cnt_num[num] += 1
        
        max_group = max(cnt_num)
        ans = [[] for _ in range(max_group)]
        for i in range(1, 201):
            for j in range(cnt_num[i]):
                ans[j].append(i)
                
        return ans

File: LeetCode/Medium/2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.rs
impl Solution {
    pub fn find_matrix(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut cnt_num = vec![0; 201];
        for num in nums {
            cnt_num[num as usize] += 1;
        }

        let max_group = cnt_num.iter().max().unwrap();
        let mut ans = vec![vec![]; *max_group as usize];
        for i in 0..201 {
            for j in 0..cnt_num[i] {
                ans[j as usize].push(i as i32);
            }
        }
        return ans;
    }
}
